plan/physical_plan_builder.go

```diff
@@ -280,6 +284,32 @@ func (p *DataSource) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlan
 	if info != nil {
 		return info, nil
 	}
+	info, err = p.tryToConvert2DummyScan(prop)
+	if info != nil || err != nil {
+		return info, errors.Trace(err)
+	}
+	indices, includeTableScan := availableIndices(p.table)
+	if includeTableScan {
+		info, err = p.convert2TableScan(prop)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	for _, index := range indices {
+		indexInfo, err := p.convert2IndexScan(prop, index)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if indexInfo.cost < info.cost {
+			info = indexInfo
+		}
+	}
+	return info, errors.Trace(p.storePlanInfo(prop, info))
+}
+
+// tryToConvert2DummyScan is an optimization which checks if its parent is a selection with a constant condition
+// that evaluates to false, if so there is no need to do real physical scan, a dummy scan will do.
```

Comment:
tryToConvert2DummyScan is an optimization which checks if its parent is a selection with a constant condition that evaluates to false. If it is, there is no need for a real physical scan, a dummy scan will do.